src/DocumentFormat.OpenXml.Framework/Packaging/OpenXmlPackage.cs

```diff
@@ -62,11 +60,20 @@ private protected OpenXmlPackage(Package package, OpenSettings? settings = null)
             }
 
             OpenSettings = new OpenSettings(settings);
-            _package = package;
 
-            Load(package);
+            var feature = new PackageImpl(package);
+
+            Features.Set<IPackageFeature>(feature);
+
+            RegisterOnClose(package.Close);
+
+            Load(feature);
         }
 
+        private List<Action>? _onClose;
```

Comment:
The `Action` is the delegate type. And we can use `+=` to add the new action. It is fast and thread safe. And I think it better than the `List<Action>` type.

```suggestion
        private Action? _onClose;
        internal void RegisterOnClose(Action action) => _onClose += action;
```